membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -96,41 +97,40 @@ object AccountDetails {
         case _ => Json.obj()
       }
 
-      def externalisePlanName(plan: RatePlan): Option[String] = plan.product match {
-        case _: Product.Weekly => if (plan.name.contains("Six for Six")) Some("currently on '6 for 6'") else None
-        case _: Product.Paper => Some(plan.name.replace("+", " plus Digital Subscription"))
+      def externalisePlanName(plan: RatePlan): Option[String] = plan.product(catalog) match {
+        case _: Product.Weekly => if (plan.name(catalog).contains("Six for Six")) Some("currently on '6 for 6'") else None
+        case _: Product.Paper => Some(plan.name(catalog).replace("+", " plus Digital Subscription"))
         case _ => None
       }
 
+      def maybePaperDaysOfWeek(plan: RatePlan) = {
+        val dayNames = for {
+          charge <- plan.ratePlanCharges.list.toList
+            .filterNot(_.pricing.isFree) // note 'Echo Legacy' rate plan has all days of week but some are zero price, this filters those out
+          catalogZuoraPlan <- catalog.catalogMap.get(plan.productRatePlanId)
+          dayName <- catalogZuoraPlan.productRatePlanCharges
+            .get(charge.productRatePlanChargeId)
+            .collect { case benefit: PaperDay => DayOfWeek.of(benefit.dayOfTheWeekIndex).getDisplayName(TextStyle.FULL, Locale.ENGLISH) }
+        } yield dayName
+
+        if (dayNames.nonEmpty) Json.obj("daysOfWeek" -> dayNames) else Json.obj()
```

Comment:
instead of pattern matching the result of the logic that was in the ChargeListReads, we have to do it all in this function
Old Reads code was here:
  https://github.com/guardian/members-data-api/blob/1433e4f1800f3b286bab67916b82fb6a7ae81e0b/membership-common/src/main/scala/com/gu/memsub/subsv2/reads/ChargeListReads.scala#L172